File: app/src/main/java/com/example/tim/communityclock/di/AppComponent.java
package com.example.tim.communityclock.di;


import com.example.tim.communityclock.App;
import com.example.tim.communityclock.ui.alarmdisplay.AlarmDisplayActivityModule;
import com.example.tim.communityclock.ui.main.MainModule;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        MainModule.class,
        AlarmDisplayActivityModule.class,
        BuildersModule.class})
public interface AppComponent {
    @Component.Builder
    interface Builder{
        @BindsInstance
        Builder application(App application);
        AppComponent build();
    }
    void inject(App app);
}

File: app/src/main/java/com/example/tim/communityclock/di/AppComponent.kt
package com.example.tim.communityclock.di


import com.example.tim.communityclock.App
import com.example.tim.communityclock.data.remote.core.RepositoryModule

import javax.inject.Singleton

import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule

@Singleton
@Component(modules = [
    AndroidInjectionModule::class,
    AppModule::class,
    ViewModelModule::class,
    BuildersModule::class,
    DatabaseModule::class,
    RepositoryModule::class
])
interface AppComponent {
    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: App): Builder

        fun build(): AppComponent
    }

    fun inject(app: App)
}